ee/tools/puppet/fleetdm/lib/puppet/util/fleet_client.rb

```diff
@@ -15,15 +15,16 @@ def initialize(host, token)
     # Pre-assigns a profile to a host. Note that the profile assignment is not
     # effective until the sibling `match_profiles` method is called.
     #
+    # @param run_identifier [String] Used to identify this run during profile matching.
```

Comment:
I presume this is the case, but just to be extra cautious - the `catalog_uuid` is not only unique to a _run_, but to a single device inside a given run, right? (maybe that's implied, that a run always targets a single device, but not being familiar with puppet I prefer to confirm!)

Reply:
thanks for asking this, I was pretty sure it's the case, but I went to double check and I found [this](https://www.puppet.com/docs/puppetdb/7/api/query/v4/catalogs.html):

> transaction_uuid (string): a string used to tie a catalog to a report from the same Puppet run (use catalog_uuid when running off a cached catalog).
> 
> catalog_uuid (string): a string used to tie a catalog to a report to the catalog used from that Puppet run.

so, there are scenarios when it's not unique. I looked into using `transaction_uuid` but it's not available to us in the custom functions we have. I will see if I can figure out a way to get it, there's a lot of magic and hidden variables.

Reply:
Would combining both `closure_scope.catalog.catalog_uuid `and `Puppet[:node_name_value]` into one value give it any more useful uniqueness?

Reply:
thanks Sarah, I think I'll probably go with that if I don't find a way to get another unique value.

having both is not ideal because we still have the same bugs described in the issue description if the same node is using a cached catalog, but it's at least better.

Reply:
@mna @gillespi314 upon more reading, when an agent uses a cached catalog, the catalog is never compiled in the server, and thus we'll never get any pre-assignments!

by virtue of that, I think our module requires a catalog compilation to work, which means an unique id!

I will still add the node name to be on the safe side.